### PR TITLE
Add tests of HTML serializer

### PR DESCRIPTION
**Pull request by eikhr**
To start work on issue #1 I wrote some tests of the HTML serializer/deserializer.

I'm not super familiar with LEGO-editor, so there are probably edge cases I haven't thought about, but I did try to test all the stuff that there is code for handling in the serializer. I also added some tests that do round trip serialization and deserialization, to check that the HTML doesn't change in unexpected ways.

While testing I found four issues that I have fixed.
- h5 elements are supported (there is even a markdown-shortcut for them) but didn't serialize to HTML.
- serialization of images produced `` tags that were a mess, with duplicate attributes, missing quotes around attribute values and slate node internals (children and type) added as attributes.
- links produced HTML tags with `target="blank"` instead of `target="_blank"` ([kind of still works](https://css-tricks.com/targetblank/), but probably not intended) and were missing quotes around href value
- code blocks added extra `<code>` tags and white-space around the code each time is was deserialized and serialized again. (read saved and loaded in LEGO)

### CODE REVIEW

File: src/serializer.test.ts [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    const link = paragraph.children[1];
+    expect(link.type).toBe('link');
+    expect(link.url).toBe('https://abakus.no');
+    expect(link.children).toHaveLength(1);
+    expect(link.children[0].text).toBe('with a link');
+  });
+
+  it('deserializes document with <br> tag', () => {
+    const deserialized = deserializeHtmlString(
+      '<p>paragraph <br> with a line break <br/> or two</p>'
```
Comment (ivarnakken) [inline src/serializer.test.ts:305]: `<br>` (without a slash) is sufficient, no? Or did you specifically want to test it with and without a slash?

File: src/serializer.test.ts [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    expect(deserialized).toHaveLength(1);
+
+    expect(deserialized[0].type).toBe('paragraph');
+    expect(deserialized[0].children).toHaveLength(1);
+    expect(deserialized[0].children[0].text).toBe('should be found');
+  });
+
+  it('deserializes an image figure', () => {
+    const deserialized = deserializeHtmlString(
+      '<figure><img src="image_src.jpg" alt="Cool figure 😎"><figcaption>Fig.1 - Cool stuff.</figcaption></figure>'
```
Comment (ivarnakken) [inline src/serializer.test.ts:384]: Missing a closing slash on the `img` here as well. But, I just realized these were optional in HTML5 on such void elements. Oh well ...

File: src/serializer.test.ts [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    expect(image.type).toBe('image');
+    expect(image.src).toBe('image_src.jpg');
+    expect(image.alt).toBe('Cool figure 😎');
+
+    expect(caption.type).toBe('image_caption');
+    expect(caption.children).toHaveLength(1);
+    expect(caption.children[0].text).toBe('Fig.1 - Cool stuff.');
+  });
+
+  it('ignores <script /> and <style />', () => {
```
Comment (ivarnakken) [inline src/serializer.test.ts:404]: Nice addition! 💯

File: src/serializer.test.ts (hunk 1 above)
Comment (eikhr) [reply src/serializer.test.ts:305]: I did want to test both, but now that I think about it it's very unnecessary as this is handled by the dom parser, and not the deserializer.

File: src/serializer.test.ts [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+  element: Node,
+  marks: {
+    bold?: boolean;
+    italic?: boolean;
+    underline?: boolean;
+    code?: boolean;
+  }
+): void => {
+  marks.bold
+    ? expect(element.bold).toBe(true)
```
[the remark this answers is not in the document]
Comment (eikhr) [reply src/serializer.test.ts]: Not quite a one-liner, but I think it's pretty good now :)

**Comment by eikhr** [comment]
> While you're in here, maybe add the missing serialization of `blockquote` as well? 😅

Absolutely! Great catch, I completely missed that
